src/validator.ts

```diff
@@ -194,47 +215,54 @@ function guessParameterValue(parameterName: string, parameter: any): string | st
     }
 
     // Check if the Ref for the parameter has been defined at runtime
-    if (parameterRuntimeOverride.hasOwnProperty(parameterName)) {
+    const parameterOverride = parameterRuntimeOverride[parameterName]
+    if (parameterOverride !== undefined) {
         // Check the parameter provided at runtime is within the allowed property list (if specified)
-        return parameterRuntimeOverride[parameterName];
+        return parameterOverride;
     } else if (parameter.hasOwnProperty('Default')) {
         // See if Default property is present and populate
         return parameter['Default'];
-    } else if (parameter.hasOwnProperty('AllowedValues') && parameter['AllowedValues'].length > 0) {
-        // See if AllowedValues has been specified
-        return parameter['AllowedValues'][0];
     } else {
-        const rawParameterType = parameter['Type'];
+        const logLevel = (okToGuess)
+            ? 'info'
+            : 'crit';
+        addError(logLevel, 'Guessing parameter value', ['Parameters', parameterName], 'Parameters')
```

Comment:
This could be misleading; if I specify `--no-guess-parameters` I would be confused to see a critical error saying `Guessing parameter value` (It shouldn't be guessing my parameter values!). Maybe something different for this use case would be better. `Value for parameter not provided`?

Reply:
Yeah that would be sensible. Does it still make sense to guess in this case (it kind of has to continue validation...)?

Reply:
I've also removed the output completely if `okToGuess` is true (so behaviour is identical to current)

Reply:
Good shout, maybe something like:
If parameter name  IN `--guess--parameters` OR guessAll => "Guessing parameter value" (INFO)
If `--no-guess-parameters` => "Value for parameter not provided, continuing validation with empty input"

Then we are enforcing what the user expects by using the CLI option.

Reply:
Can I leave off the INFO message completely? It currently works like that and would lead to lots of noise in the output. (I think I was mistaken in adding it.)